the remove breakpoint command should be unified into db-

Vane11ope opened the pull request:
in previous contribution, i made autocomplete for db- work but still help message suggest 'db -' to use, yet other commands, for example dbh- and drx- have got no space between command and '-', so i guess it is pretty safe to modify help message for "remove breakpoints" to be "db-" from "db -", and that is where my intention lies in when i made the autocomplete work for it anyway.

radare commented:
Uhm not sure about this change. Because the - should be part of the command. Even if we handle the space assuming negative addresses are not valid.. for usability, and as long as the command determines the behaviour i think it should be kept as part of the command

Vane11ope commented:
cool, i basically agree with your thought to keep "db -<addr>" help message as well.

what i thought, and what i wanted to know before looking into code base was this.

1. i wanted to delete all the bps i put at once, because i put way too many of them (to solve some ctf questions)
2. then realized db? did not show anything about it, it just showed the way to either deleting them one by one or disabling them one by one.
3. then i looked into code, and there was this "db-*" function that does the job.

therefore i thought that might be nice to notify the existence of the command.

so my idea is this.

1. let's keep "db -<addr>" help message.
2. but still, let me add "db- " <addr> help message as well, because with that, people can use autocomplete for deleting addresses.
3. and probably very kind if you let me add "db-*" as well because with that way, you can clean up all the bp messes.

what would you reckon sir?

radare commented:
> On 11 Feb 2018, at 14:17, Vanellope <notifications@github.com> wrote:
> 
> cool, i basically agree with your thought to keep "db -" help message as well.
> 
> what i thought, and what i wanted to know before looking into code base was this.
> 
> i wanted to delete all the bps i put at once, because i put way too many of them (to solve some ctf questions)
db-*
> then realized db? did not show anything about it, it just showed the way to either deleting them one by one or disabling them one by one.
> then i looked into code, and there was this "db-*" function that does the job.


This glob thing works on flags and many other commands
> therefore i thought that might be nice to notify the existence of the command.
> 
> so my idea is this.
> 
> let's keep "db -" help message.
Uhm that space is confusing imho
> but still, let me add "db- " help message as well, because with that, people can use autocomplete for deleting addresses.
> and probably very kind if you let me add "db-*" as well because with that way, you can clean up all the bp messes.
> what would you reckon sir?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Vane11ope commented:
![db](https://user-images.githubusercontent.com/29271244/36078114-07b56bdc-0fb6-11e8-836f-823bfc92407d.png)

this is what it looks like now.

and removing bp is just "db -<addr>" according to this help message. but db -* does not do anything meanwhile db-* does.

but from this help message, user can't get there because there is no help message for "db-". 
only db -<addr> exists.

so I would like to add db- help message as well and at last, i also would like to add db-* because 

db -*
db- *

does not do the job as it looks like, only db-* delete all the bps.

if you think this is too small point, then i will dismiss because i know all of those things already. 
but i think we will not lose anything if i add just two lines of help messages about db- and db-* though.

Vane11ope commented:
okay last try.

i didn't delete anything, i just added two help messages that would be of help for some users. 

if you still don't like the change then I will dismiss this and close.

XVilka commented:
Sorry, I don't undestand, why not to kill the `db -` help? I think maybe @radare misunderstood, but it is certainly against the other commands which follow the syntax `<cmd>-` for removing.

Vane11ope commented:
in the first commit, i removed db -\<addr\> help message because the command looked kind of odd anyway compared to other commands (drx-, dbh-). but i thought he did not like it so in this new commit i got that back and just added two commands help that i thought was needed (because i needed them before i started contributing to radare, or more like that actually was the reason i started contributing to radare)
what would you guys reckon. this is small thing, so im so sorry if this might bother you.

XVilka commented:
@Vane11ope please just remove it back, move those 2 commands up, where is `db -` located now and we'll merge. Sorry for the frustration. If I am wrong - I will just fix it by myself then.

Vane11ope commented:
thank you :)) im so newbie here so i dont wanna annoy you guys but just wanna make this amazing tool better that's all. anyway i will remove it back and certainly do as you said and push it again in minutes.

XVilka commented:
Thank you!
